### DISCUSSION TITLE
Entity tiles incorrectly offset

### DISCUSSION THREAD

**Discussion by jwpjrdev** [Trouv/bevy_ecs_ldtk#383] [Q&A]
This lamp entity is 3x1 in terms of tiles. It looks fine in LDtk but in game (I haven't done anything special to the entity in the code) the sprite sheet is shifted up by 1 tile, resulting in the sprite above it (a carrot in this case) to be shown, while cutting off the bottom tile.

Here's the code the entity uses. Nothing special.
```rust
#[derive(Default, Component)]
pub struct Lamp;

#[derive(Default, Bundle, LdtkEntity)]
pub struct LampBundle {
    lamp: Lamp,
    #[sprite_sheet_bundle]
    sprite_bundle: LdtkSpriteSheetBundle,
}
```
Ingame:
<img width="1470" alt="Screenshot 2024-10-17 at 17 56 08" src="https://github.com/user-attachments/assets/d11043d5-d70d-4182-8442-fb03df39c738">
In the editor:
<img width="1470" alt="Screenshot 2024-10-17 at 17 55 46" src="https://github.com/user-attachments/assets/e0a5c279-cfad-479a-af46-be2d2c72124b">
<img width="1470" alt="Screenshot 2024-10-17 at 18 07 08" src="https://github.com/user-attachments/assets/e8f949c4-2b56-46a3-89ab-739105a4c1cd">
<img width="1470" alt="Screenshot 2024-10-17 at 18 07 11" src="https://github.com/user-attachments/assets/a0b216c4-b5cf-442f-ad7b-5fdd6cb0aaa8">

**Comment by zachrburke** [comment]
Have you tried `#[sprite_sheet_bundle(no_grid)]`?

I had a similar problem when creating an entity that spanned more than one tile.  I found the platformer example was doing something similar without issue, which is where I found the above.

Adding that worked in my case.  I'm not sure what the `no_grid` parameter actually does though.

**Reply by Trouv** [reply]
Yeah, the macros kinda assume that when you want a sprite sheet bundle, you want the texture atlas to be split up into the grid that the tileset has defined on it in LDtk. However, LDtk entity visuals pull from these tilesets and can actually define an entire rectangle of tiles from the tileset as the visual for an entity. Unlike a single-tile entity visual, this doesn't really map super well onto a bevy `TextureAtlas` whose grid is equal to the tileset's grid.

So, the stop-gap solution for this is `no_grid`, which creates a new texture atlas that isn't based on the grid of the LDtk tileset, but instead just makes a texture atlas layout with exactly one element in the atlas: the rectangle of the LDtk entity visual.
